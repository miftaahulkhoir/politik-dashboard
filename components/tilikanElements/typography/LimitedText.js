import { Typography } from "antd";
import React from "react";

function LimitedText({ text, charLimit = 50 }) {
  const showElipsis = text.length > charLimit ? true : false;

  return (
    <Typography.Text style={{ width: `${charLimit - 10}ch` }} ellipsis={showElipsis ? { tooltip: text } : false}>
      {text}
    </Typography.Text>
  );
}

export default LimitedText;
