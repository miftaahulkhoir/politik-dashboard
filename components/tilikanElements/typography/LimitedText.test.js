import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import LimitedText from "./LimitedText";

// number of char in mockText: 228
const mockText =
  "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Repellendus animi, nam autem, delectus similique voluptatibus eum et, perspiciatis provident incidunt aut! Quia asperiores eveniet culpa ullam unde alias dolore corrupti.";
describe("LimitedText component", () => {
  it("should not have unexpected changes", () => {
    const tree = renderer.create(<LimitedText text={mockText} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render elipsis styles if text is longer than the char limit", () => {
    render(<LimitedText text={mockText} />);
    const textElement = screen.getByText(mockText);

    expect(textElement).toHaveClass("ant-typography-ellipsis");
  });

  it("should not render elipsis styles if text not longer than the char limit", () => {
    render(<LimitedText text={mockText} charLimit={mockText.length + 1} />);

    const textElement = screen.getByText(mockText);
    expect(textElement).not.toHaveClass("ant-typography-ellipsis");
  });
});
