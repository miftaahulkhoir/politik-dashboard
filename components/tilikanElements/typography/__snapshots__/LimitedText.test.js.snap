// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LimitedText component should not have unexpected changes 1`] = `
<span
  aria-label="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Repellendus animi, nam autem, delectus similique voluptatibus eum et, perspiciatis provident incidunt aut! Quia asperiores eveniet culpa ullam unde alias dolore corrupti."
  className="ant-typography ant-typography-ellipsis ant-typography-single-line css-dev-only-do-not-override-1n7nwfa"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    {
      "WebkitLineClamp": undefined,
      "width": "40ch",
    }
  }
>
  Lorem, ipsum dolor sit amet consectetur adipisicing elit. Repellendus animi, nam autem, delectus similique voluptatibus eum et, perspiciatis provident incidunt aut! Quia asperiores eveniet culpa ullam unde alias dolore corrupti.
  <span
    aria-hidden={true}
    style={
      {
        "display": "block",
        "fontSize": 0,
        "left": 0,
        "pointerEvents": "none",
        "position": "fixed",
        "top": 0,
        "visibility": "hidden",
        "whiteSpace": "nowrap",
        "wordBreak": "keep-all",
        "zIndex": -9999,
      }
    }
  >
    lg
  </span>
  <span
    aria-hidden={true}
    style={
      {
        "display": "block",
        "fontSize": 0,
        "left": 0,
        "margin": 0,
        "padding": 0,
        "pointerEvents": "none",
        "position": "fixed",
        "top": 0,
        "visibility": "hidden",
        "whiteSpace": "normal",
        "width": 0,
        "zIndex": -9999,
      }
    }
  >
    <span
      aria-hidden={true}
    >
      ...
    </span>
  </span>
</span>
`;
